apps/protocol-frontend/src/components/ReportForm.tsx

```diff
@@ -50,6 +61,44 @@ const ReportForm = ({ onFinish }: { onFinish: () => void }) => {
     allDaos,
   } = useUser();
 
+  const [, setIpfsUri] = useState('');
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+  const fileInputField = useRef<HTMLInputElement>(null);
+  const [fileError, setFileError] = useState<string | null>(null);
+  const [isUploading, setIsUploading] = useState(false);
+
+  const handleFileUploadButtonClick = () => {
+    if (fileInputField.current !== null) {
+      fileInputField.current.click();
+    }
+  };
+  const handleImageSelect = async (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const file = event?.target.files ? event.target.files[0] : null;
+    if (file !== null) {
+      setValue('proof', '');
+      setFileError(null);
+      if (file.size > MAX_FILE_UPLOAD_SIZE) {
+        setFileError(
+          'File size is too large. Please select something smaller than 5 MB.',
+        );
+        return;
+      }
+    }
+    setSelectedFile(file);
+  };
+
+  const handleIpfsUpload = async () => {
+    if (selectedFile) {
+      setIsUploading(true);
+      const ipfsImageUri = await uploadFileIpfs(selectedFile);
+      setIsUploading(false);
+      setIpfsUri(ipfsImageUri);
+      setValue('proof', ipfsImageUri);
+    }
+  };
```

Comment:
Prime candidate  for a hook it looks like. Especially since the logic is duplicated

Reply:
Now that the logic works with the new changes will restructure this